### PR TITLE
Add Codeengine DA

### PR DESCRIPTION
**Pull request by daniel-butler-irl**
### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by ocofaigh** [comment]
@daniel-butler-irl FYI, we are moving back to the RAG repo in https://github.com/terraform-ibm-modules/stack-retrieval-augmented-generation/pull/180 so won't merge any new PRs for the basic variation in the repo.

Once that PR is merged, I suggest you target this PR to that repo. No reason why we can't review and test it using this PR first though

File: solutions/basic/stack_definition.json [hunk 1]
```diff
@@ -371,6 +371,10 @@
           "name": "cos_region",
           "value": "ref:../../inputs/region"
         },
+        {
+          "name": "prefix",
+          "value": "ref:../../inputs/prefix"
```
Comment (ocofaigh) [inline solutions/basic/stack_definition.json:376]: oh good spot - I asked @akocbek to fix this in https://github.com/terraform-ibm-modules/stack-retrieval-augmented-generation/pull/180/files#r1738514634

**Comment by daniel-butler-irl** [comment]
Ok, are we adding the scc fix to this repo still? Pipeline won't pass without it

File: solutions/basic/stack_definition.json [hunk 2]
```diff
@@ -559,15 +592,15 @@
...
         },
         {
           "name": "cd_code_engine_project",
-          "value": "generative-ai-sample-app-cd-project"
+          "value": "ref:../Workload - Code Engine Projects/outputs/project_2_name"
         },
         {
           "name": "registry_namespace",
-          "value": "rag-sample-app"
+          "value": "rag-app"
```
Comment (ocofaigh) [inline solutions/basic/stack_definition.json:607]: Is this not a breaking change? Will the old ICR namespace be destroyed and new one created?

File: solutions/basic/stack_definition.json [hunk 3]
```diff
@@ -617,9 +650,17 @@
...
+          "name": "create_icr_namespace",
+          "value": true
+        },
         {
           "name": "add_container_name_suffix",
           "value": true
+        },
+        {
+          "name": "add_code_engine_prefix",
+          "value": false
```
Comment (ocofaigh) [inline solutions/basic/stack_definition.json:667]: Is there a reason this is `false`? @in-1911 any preference to what this should be?

Comment (daniel-butler-irl) [reply solutions/basic/stack_definition.json:667]: Yes, we already have the prefix added. This is to prevent ALM adding it a second time.

**Review by ocofaigh — COMMENTED** [review]
@daniel-butler-irl What is the impact here for consumers upgrading? Someone who will of deployed an old version of the stack will have a code engine project that is already set up with the pipeline. But if they upgrade does that get blown away and replaced by the new projects created by code engine DA? I think we need to do an upgrade test to fully understand

File: solutions/basic/stack_definition.json (hunk 2 above)
Comment (daniel-butler-irl) [reply solutions/basic/stack_definition.json:607]: for me the old name was generating a namespace over 30 character limit... I don't know how its passing in the main

**Comment by ocofaigh** [comment]
> Ok, are we adding the scc fix to this repo still? Pipeline won't pass without it

^^ If you want to run the tests then yes lets add the workaround to your PR. The workaround is to pass an empty list for profile attachements (same as [this](https://github.com/terraform-ibm-modules/dev-rag/blob/ab382817bdc2d02a39d099c4d3af444d9b1b1be0/solutions/basic/stack_definition.json#L392-L393))

**Comment by daniel-butler-irl** [comment]
I suggest this breaking change gets added into the renaming breaking change.

**Comment by ocofaigh** [comment]
Even though its a breaking change, we still need to document migration steps if possible (although there are other issues blocking us from actually being able to migrate in projects so maybe we need a call on this).
Perhaps we should include it as part of Andrej's PR so its just 1 breaking change? https://github.com/terraform-ibm-modules/stack-retrieval-augmented-generation/pull/180

**Pull request closed without merging**